tests/Integration/SDK/SpanBuilderTest.php

```diff
@@ -230,6 +242,71 @@ public function test_set_attribute_after_end(): void
         $span->end();
     }
 
+    /**
+     * @group trace-compliance
+     */
+    // public function test_set_attribute_empty_string_value_is_set(): void
+    // {
+    //     /** @var Span $span */
+    //     $span = $this
+    //         ->tracer
+    //         ->spanBuilder(self::SPAN_NAME)
+    //         ->setAttribute('nil', null)
+    //         ->setAttribute('empty-string', '')
+    //         ->startSpan();
+
+    //     $attributes = $span->toSpanData()->getAttributes();
+    //     $this->assertSame(1, $attributes->count());
+    //     $this->assertSame('', $attributes->get('empty-string'));
+    //     $this->assertNull($attributes->get('nil'));
+
+    //     $span->end();
+    // }
+
+    /**
+     * @group trace-compliance
+     */
+    public function test_set_attribute_only_null_string_value_should_not_be_set(): void
+    {
+        /** @var Span $span */
+        $span = $this
+            ->tracer
+            ->spanBuilder(self::SPAN_NAME)
+            ->setAttribute('nil', null)
+            ->startSpan();
+
+        $attributes = $span->toSpanData()->getAttributes();
+        $this->assertEmpty($span->toSpanData()->getAttributes());
+        $this->assertNull($attributes->get('nil'));
+
+        $span->end();
+    }
+
+    /**
+     * @group trace-compliance
+     */
+    public function test_set_attribute_no_effect_after_start_span(): void
+    {
+        $spanBuilder = $this->tracer->spanBuilder(self::SPAN_NAME);
+
+        /** @var Span $span */
+        $span = $spanBuilder
+            ->setAttribute('foo', 'bar')
+            ->setAttribute('bar', 123)
+            ->startSpan();
+
+        $attributes = $span->toSpanData()->getAttributes();
+        $this->assertSame(2, $attributes->count());
+
+        $spanBuilder
+            ->setAttribute('bar1', 77);
+
+        $attributes = $span->toSpanData()->getAttributes();
+        $this->assertSame(2, $attributes->count());
```

Comment:
See my comment in slack. This is not really testing if adding an attribute is not possible, but if it changes the number of attributes.
It would be nice, if you could change it, and also the other tests, which only test for attribute count.